### ISSUE TITLE
Packaging for the AUR

### ISSUE THREAD

**Issue by PolarianDev** [Spotifyd/spotifyd#1165]
Hello,

I am interested in putting this onto the AUR and keeping it supported. I have a few suggestions as well.

- Please link the build instructions within the README.md (I might submit a PR to add this), as http://spotifyd.github.io/spotifyd/installation/index.html took me a bit to find :)

- Be more clear about the releases, what are they? Binaries or just source code? I have not figured that one out yet :/

- Please provide a list of dependencies which spotifyd relies on, otherwise I am just guessing from looking at the install guides for other distributions.

I also recommend, providing install links within the README.md, as currently its a little obscure, all it has to be is a hyperlink to the wiki page (hosted on github) for that specific operating system/linux distribution, it would help a lot :)

Hope these advice help, however I do need a list of dependencies before I can make the AUR package!

I have already checked the AUR, spotifyd is not packaged for it.

**Comment by PolarianDev** [comment]
My mistake, it is within the official arch linux repository, thus does not need to be added to the AUR.

Please still take some of the advice above however :)

**Comment by PolarianDev** [comment]
Definitely should add to the docs "installing on arch linux" section :)

**Comment by eladyn** [comment]
Yeah, thanks for raising these. I think, all of them are valid points, the wiki is still quite inconsistent and could be much more polished in many places.

Help with this would be greatly appreciated! Otherwise, we'll see, how we can slowly improve things over time. :)

**Comment by PolarianDev** [comment]
Sure, I am happy to contribute when I get time :)

**Comment by slondr** [comment]
Just in case this info is still unclear:

> Be more clear about the releases, what are they? Binaries or just source code? I have not figured that one out yet :/

`Source code` is the code, everything else are prebuilt binaries.

> Please provide a list of dependencies which spotifyd relies on, otherwise I am just guessing from looking at the install guides for other distributions.

I believe the minimum viable runtime dependencies set for spotifyd is libogg. You will also almost certainly want to play sound which requires a compatible audio sink (this varies based on system configuration; Arch will likely push you towards either Alsa or Pulse, with the latter being the most populate), and a lot of functionality is tied to dbus too. That's how you end up with the 4 packages Arch defines as dependencies :)

I think we should catalog the easiest way to get up and running with spotifyd on various platforms in the Readme; that info should probably be the most discoverable info users are presented with. 

I'd suggest that we include system package manager installation instructions for distros which ship up-to-date versions (directing users to install old versions will likely lead to users experiencing bugs which have already been patched). Referencing [repology](https://repology.org/project/spotifyd/versions), that includes:

- Arch and friends (incl. Arch Linux ARM, Parabola, Manjaro, Artix, etc)
- Homebrew
- Nix
- Void
- Funtoo

And that's it! For all other platforms, we should probably include a blurb about installing rustup and using `cargo install`, then directing to the https://docs.spotifyd.rs page for future reading.

Thoughts?

**Comment by PolarianDev** [comment]
Seems like someone doesn't like Ubuntu/Debian 📦 

Most developers normally only distribute `.deb` packages xD

It does sound like a good idea though, to include better supported distributions. The only thing I got to say is that Arch forks sometimes use the Arch repositories, but some also use their own, so this can be difficult to determine whether they would be installed through pacman or not :/

**Comment by PolarianDev** [comment]
Pretty much no, Manjaro tries to convince users to use their wrapper (pamac) which also is a AUR helper, but in general any arch derivative should contain pacman, they will be able to pull the package in the same way:

`pacman -Syu spotifyd`

the only issue is that some derivatives are not using the official arch repository (like Manjaro), these may not contain spotifyd and thus pacman will fail to find them.

However, in general, it should work for all derivatives (most of them use the official repository), see: https://archlinux.org/packages/community/x86_64/spotifyd/
